ionoscloud/data_source_backup_unit.go

```diff
@@ -0,0 +1,144 @@
+package ionoscloud
+
+import (
+	"context"
+	"fmt"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	ionoscloud "github.com/ionos-cloud/sdk-go/v5"
+	"strings"
+)
+
+func dataSourceBackupUnit() *schema.Resource {
+	return &schema.Resource{
+		ReadContext: dataSourceBackupUnitRead,
+		Schema: map[string]*schema.Schema{
+			"id": {
+				Type:        schema.TypeString,
+				Description: "Id of the backup unit.",
+				Optional:    true,
+			},
+			"name": {
+				Type:        schema.TypeString,
+				Description: "Alphanumeric name you want assigned to the backup unit.",
+				Optional:    true,
+			},
+			"email": {
+				Type:        schema.TypeString,
+				Description: "The e-mail address you want assigned to the backup unit.",
+				Computed:    true,
+			},
+			"login": {
+				Type:        schema.TypeString,
+				Description: "The login associated with the backup unit. Derived from the contract number",
+				Computed:    true,
+			},
+		},
+		Timeouts: &resourceDefaultTimeouts,
+	}
+}
+
+func dataSourceBackupUnitRead(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	client := meta.(*ionoscloud.APIClient)
+
+	id, idOk := d.GetOk("id")
+	name, nameOk := d.GetOk("name")
+
+	if idOk && nameOk {
+		return diag.FromErr(fmt.Errorf("id and name cannot be both specified in the same time"))
+	}
+	if !idOk && !nameOk {
+		return diag.FromErr(fmt.Errorf("please provide either the backup unit id or name"))
+	}
+	var backupUnit ionoscloud.BackupUnit
+	var err error
+
+	if idOk {
+		/* search by ID */
+		backupUnit, _, err = client.BackupUnitApi.BackupunitsFindById(ctx, id.(string)).Execute()
+		if err != nil {
+			return diag.FromErr(fmt.Errorf("an error occurred while fetching the backup unit %s: %s", id.(string), err))
+		}
+	} else {
+		/* search by name */
+		var backupUnits ionoscloud.BackupUnits
+
+		backupUnits, _, err := client.BackupUnitApi.BackupunitsGet(ctx).Execute()
```

Comment:
Let's try to use a `filter.name={name}&depth=1` GET query here. Not sure the sdk will accept a `filter.name` param though, since it's not in the swagger spec.